docs/retries.md

```diff
@@ -0,0 +1,267 @@
+# Retries
```

Comment:
Let's use the term `Client Side Retries`, as customers may think these info are passed to the backend services, which happened in the past.

Reply:
Sounds good, will update.